File: python/calculate_mode_spec.py
# Write your own unit tests here!
from calculate_mode import calculate_mode
import unittest

class ModeTest(unittest.TestCase):

    def test_num_list(self):
        self.assertEqual(calculate_mode([1,2,3,3]), [3])
        self.assertEqual(calculate_mode([4.5, 0, 0]), [0])
        self.assertEqual(calculate_mode([1.5, -1, 1, 1.5]), [1.5])
        self.assertEqual(calculate_mode([1,1,2,2]), [1,2])
        self.assertEqual(calculate_mode([1,2,3]), [1,2,3])

    def test_string_list(self):
        self.assertEqual(calculate_mode(["who", "what", "where", "who"]), ['who'])
        self.assertEqual(calculate_mode(["hi", "world", "hi", "morning"]), ['hi'])

if __name__ == '__main__':
    unittest.main()

File: js/calculateModeSpec.js
// Write your own unit tests here!
var cm = require("./calculateMode");
var shallowEqualArrays = require('shallow-equal/arrays');

console.log(shallowEqualArrays(cm.calculateMode([1,2,3,3]),['3']))
console.log(JSON.stringify(cm.calculateMode([4.5, 0, 0])) == JSON.stringify(['0']))
console.log(shallowEqualArrays(cm.calculateMode([1.5, -1, 1, 1.5]),['1.5']))
console.log(shallowEqualArrays(cm.calculateMode([1,1,2,2]), ['1','2']))
console.log(shallowEqualArrays(cm.calculateMode([1,2,3]), ['1','2','3']))

console.log(shallowEqualArrays(cm.calculateMode(["who", "what", "where", "who"]), ['who']))
console.log(shallowEqualArrays(cm.calculateMode(["hi", "world", "hi", "morning"]), ['hi']))